.github/workflows/check-pr.yml

```diff
@@ -26,3 +29,148 @@ jobs:
         run: npm ci
       - name: Run validation and tests
         run: npm pack
+
+      - name: Trigger Workflows
+        uses: peter-evans/repository-dispatch@v2
+        with:
+          token: ${{ secrets.WORKFLOW_TRIGGER_SECRET }}  # or use a custom token with proper permissions
+          repository: rdkcentral/firebolt-apis 
+          event-type: trigger-workflow
+          client-payload: '{"OPENRPC_PR_BRANCH": "${{ github.event.pull_request.head.ref }}"}'
+
+      - name: Store the workflow run ID
+        run: |
+          echo "Waiting for Repo2 to finish execution"
+  wait-for-sdk-generation:
+    needs: release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Poll Firebolt-api for JavaScript SDK generation
+        id: poll-javascript-sdk
+        run: |
+          TOKEN="${{ secrets.WORKFLOW_TRIGGER_SECRET }}"
+          REPO_OWNER="rdkcentral"
+          REPO_NAME="firebolt-apis"
+          WORKFLOW_NAME="MFOS standalone sanity report - CORE,MANAGE,DISCOVERY"
+
+          while true; do
+            response=$(curl -s -H "Authorization: token $TOKEN" \
+              "https://api.github.com/repos/$REPO_OWNER/$REPO_NAME/actions/workflows" | jq \
+              --arg WORKFLOW_NAME "$WORKFLOW_NAME" \
+              '.workflows[] | select(.name == $WORKFLOW_NAME)')
+            
+            workflow_id=$(echo $response | jq -r '.id')
+            if [ "$workflow_id" == "null" ]; then
+              echo "Workflow not found, retrying in 10 seconds..."
+              sleep 10
+              continue
+            fi
+
+            # Get the latest workflow run
+            run_response=$(curl -s -H "Authorization: token $TOKEN" \
```

Comment:
What happens if this workflow is triggered by a "firebolt-apis" PR while this is running? Will it get that one or the one that was triggered by this workflow?

Reply:
It gets the workid of the workflow that is trigged by openRPC.Immediately after triggering the workflow we're getting the work flow id from the response.